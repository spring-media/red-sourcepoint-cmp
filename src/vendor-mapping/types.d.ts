export type VendorMappingName = 'facebook' | 'twitter' | 'instagram' | 'youtube';

export type PurposeMappingName = 'social';
