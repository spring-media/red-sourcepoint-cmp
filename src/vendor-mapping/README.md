# Static Vendor Mapping Package

As long as Sourcepoint doesn't provides an API for getting relationships between vendors and purposes, 
we need to have some kind of static mapping between these two entities.

## API

> The following examples refer to the esm-bundle. 
> When using the browser-bundle, all methods are properties of the global object named **RedCMP**

### `getVendorIdByName(name: string): string | undefined`

Returns the vendor id by given name or undefined if not exists.

<details>
<summary>Example</summary>
    
```javascript
import { getVendorIdByName, VENDOR_NAME_FACEBOOK } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(getVendorIdByName(VENDOR_NAME_FACEBOOK)); // 5e716fc09a0b5040d575080f
```    
</details>

### `getPurposeIdByName(name: string): string | undefined`

Returns the purpose id by given name or undefined if not exists.

<details>
<summary>Example</summary>
    
```javascript
import { getPurposeIdByName, PURPOSE_NAME_SOCIAL } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(getPurposeIdByName(PURPOSE_NAME_SOCIAL)); // 5e7adb1ee30e7d1bc1ec0252
```    
</details>

### `getVendorNameById(id: string): string | undefined`

Returns the vendor name by given id or undefined if not exists.

<details>
<summary>Example</summary>
    
```javascript
import { getVendorNameById, VENDOR_ID_FACEBOOK } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(getVendorNameById(VENDOR_ID_FACEBOOK)); // facebook
```    
</details>

### `getPurposeNameById(id: string): string | undefined`

Returns the purpose name by given id or undefined if not exists.

<details>
<summary>Example</summary>
    
```javascript
import { getPurposeNameById, PURPOSE_ID_SOCIAL } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(getPurposeNameById(PURPOSE_ID_SOCIAL)); // 5e7adb1ee30e7d1bc1ec0252
```    
</details>

### `hasRelationByName(vendorName: string, purposeName: string): boolean`

Returns true if given vendor(name) has a relation to given purpose(name), otherwise false.

<details>
<summary>Example</summary>
    
```javascript
import { hasRelationByName, VENDOR_NAME_INSTAGRAM, PURPOSE_NAME_SOCIAL } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(hasRelationByName(VENDOR_NAME_INSTAGRAM, PURPOSE_NAME_SOCIAL)); // true
```
</details>

### `hasRelationById(vendorId: string, purposeId: string): boolean`

Returns true if given vendor(id) has a relation to given purpose(id), otherwise false.

<details>
<summary>Example</summary>
    
```javascript
import { hasRelationById, VENDOR_ID_INSTAGRAM, PURPOSE_ID_SOCIAL } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(hasRelationById(VENDOR_ID_INSTAGRAM, PURPOSE_ID_SOCIAL)); // true
```
</details>

### `getVendorPurposesByName(vendorName: string): string[]`

Returns the purposes (names) for which given vendor has a relationship.

<details>
<summary>Example</summary>
    
```javascript
import { getVendorPurposesByName, VENDOR_NAME_INSTAGRAM } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(getVendorPurposesByName(VENDOR_NAME_INSTAGRAM)); // ['social']
```
</details>

### `getVendorPurposesById(vendorId: string): string[]`

Returns the purposes (ids) for which given vendor has a relationship.

<details>
<summary>Example</summary>
    
```javascript
import { getVendorPurposesById, VENDOR_ID_INSTAGRAM } from '@spring-media/red-sourcepoint-cmp/dist/esm/vendor-mapping';

console.log(getVendorPurposesById(VENDOR_ID_INSTAGRAM)); // ['5e7adb1ee30e7d1bc1ec0252']
```
</details>
