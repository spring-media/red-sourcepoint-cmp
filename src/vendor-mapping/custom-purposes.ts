export const PURPOSE_NAME_SOCIAL = 'social';
export const PURPOSE_ID_SOCIAL = '5e7adb1ee30e7d1bc1ec0252';

export const customPurposes: Map<string, string> = new Map([
  [PURPOSE_ID_SOCIAL, PURPOSE_NAME_SOCIAL],
  [PURPOSE_NAME_SOCIAL, PURPOSE_ID_SOCIAL],
]);
