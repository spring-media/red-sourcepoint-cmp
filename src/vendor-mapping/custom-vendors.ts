export const VENDOR_NAME_FACEBOOK = 'facebook';
export const VENDOR_ID_FACEBOOK = '5ee91b9593fc094b59242e25';

export const VENDOR_NAME_TWITTER = 'twitter';
export const VENDOR_ID_TWITTER = '5e71760b69966540e4554f01';

export const VENDOR_NAME_INSTAGRAM = 'instagram';
export const VENDOR_ID_INSTAGRAM = '5e717c8e69966540e4554f05';

export const VENDOR_NAME_YOUTUBE = 'youtube';
export const VENDOR_ID_YOUTUBE = '5e7ac3fae30e7d1bc1ebf5e8';

export const VENDOR_NAME_SOUNDCLOUD = 'soundcloud';
export const VENDOR_ID_SOUNDCLOUD = '5e7179e49a0b5040d5750812';

export const VENDOR_NAME_REDDIT = 'reddit';
export const VENDOR_ID_REDDIT = '5e7f6927b8e05c4e491e7381';

export const VENDOR_NAME_PODIGEE = 'podigee';
export const VENDOR_ID_PODIGEE = '5e717ad49a0b5040d5750815';
