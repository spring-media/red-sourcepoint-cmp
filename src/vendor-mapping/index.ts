export * from './vendors';
export * from './purposes';
export * from './api';
