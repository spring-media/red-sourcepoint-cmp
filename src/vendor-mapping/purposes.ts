import { PurposeMappingName } from './types';

export const PURPOSE_NAME_SOCIAL = 'social';
export const PURPOSE_ID_SOCIAL = '5e7adb1ee30e7d1bc1ec0252';

export const purposeMapById: Map<string, PurposeMappingName> = new Map([[PURPOSE_ID_SOCIAL, PURPOSE_NAME_SOCIAL]]);

export const purposeMapByName: Map<PurposeMappingName, string> = new Map([[PURPOSE_NAME_SOCIAL, PURPOSE_ID_SOCIAL]]);
