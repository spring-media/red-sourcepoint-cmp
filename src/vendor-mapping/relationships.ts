import { PURPOSE_ID_SOCIAL, PURPOSE_NAME_SOCIAL } from './custom-purposes';
import {
  VENDOR_ID_FACEBOOK,
  VENDOR_ID_INSTAGRAM,
  VENDOR_ID_PODIGEE,
  VENDOR_ID_REDDIT,
  VENDOR_ID_SOUNDCLOUD,
  VENDOR_ID_TWITTER,
  VENDOR_ID_YOUTUBE,
  VENDOR_NAME_FACEBOOK,
  VENDOR_NAME_INSTAGRAM,
  VENDOR_NAME_PODIGEE,
  VENDOR_NAME_REDDIT,
  VENDOR_NAME_SOUNDCLOUD,
  VENDOR_NAME_TWITTER,
  VENDOR_NAME_YOUTUBE,
} from './custom-vendors';

export const relationships: Map<string, string[]> = new Map([
  [
    PURPOSE_ID_SOCIAL,
    [
      VENDOR_ID_FACEBOOK,
      VENDOR_ID_YOUTUBE,
      VENDOR_ID_INSTAGRAM,
      VENDOR_ID_TWITTER,
      VENDOR_ID_SOUNDCLOUD,
      VENDOR_ID_REDDIT,
      VENDOR_ID_PODIGEE,
    ],
  ],
  [
    PURPOSE_NAME_SOCIAL,
    [
      VENDOR_NAME_FACEBOOK,
      VENDOR_NAME_YOUTUBE,
      VENDOR_NAME_INSTAGRAM,
      VENDOR_NAME_TWITTER,
      VENDOR_NAME_SOUNDCLOUD,
      VENDOR_NAME_REDDIT,
      VENDOR_NAME_PODIGEE,
    ],
  ],
  [VENDOR_ID_FACEBOOK, [PURPOSE_ID_SOCIAL]],
  [VENDOR_ID_TWITTER, [PURPOSE_ID_SOCIAL]],
  [VENDOR_ID_YOUTUBE, [PURPOSE_ID_SOCIAL]],
  [VENDOR_ID_INSTAGRAM, [PURPOSE_ID_SOCIAL]],
  [VENDOR_ID_SOUNDCLOUD, [PURPOSE_ID_SOCIAL]],
  [VENDOR_ID_REDDIT, [PURPOSE_ID_SOCIAL]],
  [VENDOR_ID_PODIGEE, [PURPOSE_ID_SOCIAL]],
  [VENDOR_NAME_FACEBOOK, [PURPOSE_NAME_SOCIAL]],
  [VENDOR_NAME_TWITTER, [PURPOSE_NAME_SOCIAL]],
  [VENDOR_NAME_YOUTUBE, [PURPOSE_NAME_SOCIAL]],
  [VENDOR_NAME_INSTAGRAM, [PURPOSE_NAME_SOCIAL]],
  [VENDOR_NAME_SOUNDCLOUD, [PURPOSE_NAME_SOCIAL]],
  [VENDOR_NAME_REDDIT, [PURPOSE_NAME_SOCIAL]],
  [VENDOR_NAME_PODIGEE, [PURPOSE_NAME_SOCIAL]],
]);
