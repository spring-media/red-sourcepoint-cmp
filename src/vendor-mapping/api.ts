import { customVendors } from './custom-vendors';
import { customPurposes } from './custom-purposes';
import { relationships } from './relationships';

export const getCustomVendor = (key: string): string | undefined => customVendors.get(key);

export const getCustomPurpose = (key: string): string | undefined => customPurposes.get(key);

export const addCustomVendor = (key: string, value: string): Map<string, string> => customVendors.set(key, value);

export const addCustomPurpose = (key: string, value: string): Map<string, string> => customPurposes.set(key, value);

export const removeCustomVendor = (key: string): boolean => customVendors.delete(key);

export const removeCustomPurpose = (key: string): boolean => customPurposes.delete(key);

export const getRelations = (key: string): string[] | undefined => relationships.get(key);

export const hasRelations = (key: string): boolean => Boolean(relationships.get(key));

export const setRelations = (key: string, relations: string[]): Map<string, string[]> =>
  relationships.set(key, relations);
