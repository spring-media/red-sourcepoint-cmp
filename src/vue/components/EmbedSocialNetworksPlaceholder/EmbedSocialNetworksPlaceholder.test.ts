import Vue from 'vue';
import { mount } from '@vue/test-utils';
import { EmbedSocialNetworksPlaceholder } from './';
import { getRelations, PURPOSE_ID_SOCIAL } from '../../../vendor-mapping';

const customConsent = jest.fn();

const consentActionsStub = Vue.extend({
  name: 'ConsentActions',
  render() {
    return (
      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
      this.$scopedSlots.default!({
        customConsent,
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
      }) as any
    );
  },
});

describe('EmbedSocialNetworksPlaceholder', () => {
  it('should render without any errors', () => {
    expect(mount(EmbedSocialNetworksPlaceholder, { propsData: { privacyManagerId: 12345 } }).element).toMatchSnapshot();
  });

  it('should give a consent to the purpose social media', () => {
    const wrapper = mount(EmbedSocialNetworksPlaceholder, {
      stubs: { ConsentActions: consentActionsStub },
      propsData: { privacyManagerId: 1234 },
    });

    wrapper.find('.embed-placeholder__button').trigger('click');

    expect(customConsent).toHaveBeenCalledWith({
      purposeIds: [PURPOSE_ID_SOCIAL],
      vendorIds: getRelations(PURPOSE_ID_SOCIAL),
    });
  });
});
