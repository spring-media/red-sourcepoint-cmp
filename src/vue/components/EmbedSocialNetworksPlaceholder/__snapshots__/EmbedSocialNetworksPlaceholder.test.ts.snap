// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedSocialNetworksPlaceholder should render without any errors 1`] = `
<div
  class="embed-placeholder__container"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="84"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="16"
        cy="16"
        fill="#E9ECEF"
        r="16"
      />
       
      <rect
        fill="#E9ECEF"
        height="10"
        width="60"
        x="40"
        y="7"
      />
       
      <rect
        fill="#E9ECEF"
        height="6"
        width="40"
        x="40"
        y="19"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="100%"
        y="44"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="100%"
        y="60"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="75%"
        y="76"
      />
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
    
      An dieser Stelle findest du Inhalte aus sozialen Netzwerken
    
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
      Um mit Inhalten aus sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir deine Zustimmung.
    
  </div>
   
  <button
    class="embed-placeholder__button"
  >
    
    soziale Netzwerke aktivieren
  
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir Inhalte von Drittanbietern angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Dazu ist ggf. die Speicherung von Cookies auf
        deinem Gerät notwendig. Mehr Informationen dazu findest du
        
    <a
      class="embed-placeholder__text-link"
      href="#"
      rel="noopener"
      target="_blank"
    >
      hier
    </a>
    .
      
  </div>
</div>
`;
